File: src/main/java/ru/ioleksiv/telegram/bot/core/model/telegram/method/group/SetChatStickerSet.java
package ru.ioleksiv.telegram.bot.core.model.telegram.method.group;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.ioleksiv.telegram.bot.core.controller.network.Networker;
import ru.ioleksiv.telegram.bot.core.model.telegram.method.interfaces.ChatAction;
import ru.ioleksiv.telegram.bot.core.model.telegram.responses.CommonResponse;
import ru.ioleksiv.telegram.bot.core.model.telegram.responses.ResponseCollection;

/**
 * @see <a href="https://core.telegram.org/bots/api#setchatstickerset">setChatStickerSet</a>
 */
public class SetChatStickerSet extends ChatAction<Boolean> {
    private static final String METHOD = "setChatStickerSet";
    /**
     * sticker_set_name	String Name of the sticker set to be set as the group sticker set
     */
    @JsonProperty("sticker_set_name")
    private String stickerSetName = null;

    protected SetChatStickerSet(Networker networker) {
        super(METHOD, networker);
    }

    public String getStickerSetName() {
        return stickerSetName;
    }

    public SetChatStickerSet setStickerSetName(String stickerSetName) {
        this.stickerSetName = stickerSetName;
        return this;
    }

    @Override
    public Class<? extends CommonResponse<Boolean>> getResultWrapperClass() {
        return ResponseCollection.BooleanResponse.class;
    }

    @Override
    public SetChatStickerSet setChatId(Long chatId) {
        pSetChatId(chatId);
        return this;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/method/group/SetChatStickerSet.kt
package io.github.oleksivio.tl.kbot.server.api.method.group

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.model.ApiDict
import io.github.oleksivio.tl.kbot.server.api.model.BooleanResponse
import io.github.oleksivio.tl.kbot.server.api.model.method.common.ChatAction
import io.github.oleksivio.tl.kbot.server.api.objects.ChatId

/**
 * @see [setChatStickerSet](https://core.telegram.org/bots/api/#setchatstickerset)
 */
data class SetChatStickerSet(
    /**
     * chat_id Integer or String Yes Unique identifier for the target chat or username of the target channel
     */
    @JsonProperty(ApiDict.CHAT_ID_KEY)
    override val chatId: ChatId,
    /**
     * sticker_set_name String Name of the sticker set to be set as the group sticker set
     */
    @JsonProperty(ApiDict.STICKER_SET_NAME_KEY)
    val stickerSetName: String
) : ChatAction<Boolean>() {

    @JsonIgnore
    override val resultWrapperClass = BooleanResponse::class

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "setChatStickerSet"
}
